src/core/normalizer.coffee

```diff
@@ -159,15 +159,28 @@ class Normalizer
         if node.nextSibling?
           nodes.push(node.nextSibling)
 
+      if node.tagName == dom.DEFAULT_INLINE_TAG and !node.hasAttributes()
+        dom(node).unwrap()
+
       # Merge similar nodes
       if node.previousSibling? and node.tagName == node.previousSibling.tagName
         if _.isEqual(dom(node).attributes(), dom(node.previousSibling).attributes())
           nodes.push(node.firstChild)
           dom(node.previousSibling).merge(node)
 
-  # Alphabetize nesting order of tag names (prefer <a> to be outer-most)
+  # If a <span>, merge with the parent span if possible,
+  # otherwise, alphabetize nesting order of tag names (prefer <a> to be outer-most)
   @optimizeNesting: (node, root) ->
-    if node.parentNode.tagName > node.tagName
+    if node.tagName == dom.DEFAULT_INLINE_TAG and node.parentNode.tagName == dom.DEFAULT_INLINE_TAG
```

Comment:
Maybe it'd be more generic and still useful to just merge child + parent if the tag name matches? I can't imagine why it'd be useful to have something like `<b><b>X</b></b>`, for any given tag.